tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersAssignments.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Linq;
+using MongoDB.Analyzer.Tests.Common.DataModel;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace MongoDB.Analyzer.Tests.Common.TestCases.Builders
```

Comment:
Good job with the tests. Few comments:

I think with can merge the tests with `BuildersBasic.cs`. Lets move over these tests:

`Single_expression_variable_reassignment`
`Four_variables_reassignment` -> rename to `Multiple_expression_variables_reassignment`
`Double_variable_reassignment` -> rename to `Multiple_expression_variables_declaration_and_reassignment` and extend it to:

```
var x = Builders<User>.Filter.Eq(u => u.Address, "1");
var y = Builders<User>.Filter.Eq(u => u.Address, "2");
FilterDefinition<User> z = Builders<User>.Filter.Eq(u => u.Address, "3"), w = (Builders<User>.Filter.Eq(u => u.Address, "4"));

x = y = Builders<User>.Filter.Lt(u => u.Age, 15) & Builders<User>.Filter.Gt(u => u.Age, 65);
x = z = w = y = x = z = w = y = Builders<User>.Filter.Lt(u => u.Age, 17) & Builders<User>.Filter.Gt(u => u.Age, 18);

```

Reply:
Merged with BuildersBasic.cs